Tests/SwiftAlgorithmsTests/TestUtilities.swift

```diff
@@ -62,7 +63,25 @@ func XCTAssertEqualSequences<S1: Sequence, S2: Sequence>(
   _ message: @autoclosure () -> String = "",
   file: StaticString = #file, line: UInt = #line
 ) rethrows where S1.Element == S2.Element {
-  try XCTAssert(expression1().elementsEqual(expression2(), by: areEquivalent), message(), file: file, line: line)
+    var iter1 = try expression1().makeIterator()
+    var iter2 = try expression2().makeIterator()
+    while true {
+      switch (iter1.next(), iter2.next()) {
+      case let (e1?, e2?):
+        XCTAssertTrue(
+          areEquivalent(e1, e2),
+          "Element \(e1) in first sequence not equal to \(e2) in second sequence.",
+          file: file, line: line)
+        return
```

Comment:
This `return` looks out of place here, we only want to return from the other cases.

Reply:
I was wondering if doing like this, for two sequences like `[1,1,1,1,1,1,1]` and `[2,2,2,2,2,2,2]` we would show a lot of assertion errors(one for each mismatched element). Maybe a better idea is to collect all the diff on the sequences and show them at the end in a single assertion at the end?

Reply:
Oh yeah, good point. I guess we do want to return there, but only when the elements aren't equal — that is, if we stick to the strategy of only reporting the first mismatch.

Reply:
Speaking of testing this function, that was a bug, it was only checking the first element! Ops.